network-store-server/src/main/java/com/powsybl/network/store/server/NetworkStoreRepository.java

```diff
@@ -881,78 +943,189 @@ public void deleteSwitch(UUID networkUuid, int variantNum, String switchId) {
+
+        if (twoWindingsTransformer.isPresent()) {
+            List<TemporaryLimitAttributes> temporaryLimits = getTemporaryLimits(networkUuid, variantNum, EQUIPMENT_ID, twoWindingsTransformerId);
+
+            insertTemporaryLimitsInEquipments(List.of(twoWindingsTransformer.get()), temporaryLimits);
+        }
+        return twoWindingsTransformer;
     }
 
     public List<Resource<TwoWindingsTransformerAttributes>> getTwoWindingsTransformers(UUID networkUuid, int variantNum) {
-        return getIdentifiables(networkUuid, variantNum, mappings.getTwoWindingsTransformerMappings());
+        List<Resource<TwoWindingsTransformerAttributes>> twoWindingsTransformers = getIdentifiables(networkUuid, variantNum, mappings.getTwoWindingsTransformerMappings());
+
+        List<TemporaryLimitAttributes> temporaryLimits = getTemporaryLimits(networkUuid, variantNum, EQUIPMENT_TYPE, ResourceType.TWO_WINDINGS_TRANSFORMER.toString());
+
+        insertTemporaryLimitsInEquipments(twoWindingsTransformers, temporaryLimits);
+
+        return twoWindingsTransformers;
     }
 
     public List<Resource<TwoWindingsTransformerAttributes>> getVoltageLevelTwoWindingsTransformers(UUID networkUuid, int variantNum, String voltageLevelId) {
-        return getIdentifiablesInVoltageLevel(networkUuid, variantNum, voltageLevelId, mappings.getTwoWindingsTransformerMappings());
+        List<Resource<TwoWindingsTransformerAttributes>> twoWindingsTransformers = getIdentifiablesInVoltageLevel(networkUuid, variantNum, voltageLevelId, mappings.getTwoWindingsTransformerMappings());
+
+        // Because there are not so many two windings transformers for a specific voltageLevelId, we can search their
+        // temporary limits by their IDs instead of by the two windings transformer type.
+        ArrayList<String> equipmentsIds = new ArrayList<>();
+        for (Resource<TwoWindingsTransformerAttributes> equipment : twoWindingsTransformers) {
+            equipmentsIds.add(equipment.getId());
+        }
+        List<TemporaryLimitAttributes> temporaryLimits = getTemporaryLimitsWithInClause(networkUuid, variantNum, EQUIPMENT_ID, equipmentsIds);
+
+        insertTemporaryLimitsInEquipments(twoWindingsTransformers, temporaryLimits);
+
+        return twoWindingsTransformers;
     }
 
     public void updateTwoWindingsTransformers(UUID networkUuid, List<Resource<TwoWindingsTransformerAttributes>> resources) {
         updateIdentifiables(networkUuid, resources, mappings.getTwoWindingsTransformerMappings());
+
+        // To update the twowindingstransformer's temporary limits, we will first delete them, then create them again.
+        // This is done this way to prevent issues in case the temporary limit's primary key is to be
+        // modified because of the updated equipment's new values.
+        for (Resource<TwoWindingsTransformerAttributes> resource : resources) {
```

Comment:
Would it be interessant to have a single delete query?

Reply:
To do in a later ticket.